### ISSUE TITLE
panic: reflect: call of reflect.Value.CanInterface on zero Value

### ISSUE THREAD

**Issue by dgonyeo** [kylelemons/godebug#12]
Got it to panic:
```
panic: reflect: call of reflect.Value.CanInterface on zero Value [recovered]
	panic: reflect: call of reflect.Value.CanInterface on zero Value

goroutine 6 [running]:
panic(0x88f780, 0xc4200d4b00)
	/nix/store/1blrc2bxdfjrk3kl46l75dw4i1lpswp5-go-1.7.1/share/go/src/runtime/panic.go:500 +0x1ae
testing.tRunner.func1(0xc42009af00)
	/nix/store/1blrc2bxdfjrk3kl46l75dw4i1lpswp5-go-1.7.1/share/go/src/testing/testing.go:579 +0x474
panic(0x88f780, 0xc4200d4b00)
	/nix/store/1blrc2bxdfjrk3kl46l75dw4i1lpswp5-go-1.7.1/share/go/src/runtime/panic.go:458 +0x271
reflect.Value.CanInterface(0x0, 0x0, 0x0, 0xc41fff95a8)
	/nix/store/1blrc2bxdfjrk3kl46l75dw4i1lpswp5-go-1.7.1/share/go/src/reflect/value.go:896 +0xca
github.com/coreos/ignition/internal/vendor/github.com/kylelemons/godebug/pretty.(*Config).val2node(0xcd87e0, 0x0, 0x0, 0x0, 0x0, 0x420f58)
	/home/derek/go/src/github.com/coreos/ignition/gopath/src/github.com/coreos/ignition/internal/vendor/github.com/kylelemons/godebug/pretty/reflect.go:99 +0xe38
github.com/coreos/ignition/internal/vendor/github.com/kylelemons/godebug/pretty.(*Config).fprint(0xcd87e0, 0xc420026230, 0xc4200e3098, 0x1, 0x1)
	/home/derek/go/src/github.com/coreos/ignition/gopath/src/github.com/coreos/ignition/internal/vendor/github.com/kylelemons/godebug/pretty/public.go:61 +0xb8
github.com/coreos/ignition/internal/vendor/github.com/kylelemons/godebug/pretty.(*Config).Sprint(0xcd87e0, 0xc4200e3098, 0x1, 0x1, 0x0, 0x0)
	/home/derek/go/src/github.com/coreos/ignition/gopath/src/github.com/coreos/ignition/internal/vendor/github.com/kylelemons/godebug/pretty/public.go:83 +0x7a
github.com/coreos/ignition/internal/vendor/github.com/kylelemons/godebug/pretty.(*Config).Compare(0xcd87e0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/derek/go/src/github.com/coreos/ignition/gopath/src/github.com/coreos/ignition/internal/vendor/github.com/kylelemons/godebug/pretty/public.go:114 +0xcb
github.com/coreos/ignition/internal/vendor/github.com/kylelemons/godebug/pretty.Compare(0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/derek/go/src/github.com/coreos/ignition/gopath/src/github.com/coreos/ignition/internal/vendor/github.com/kylelemons/godebug/pretty/public.go:106 +0x77
github.com/coreos/ignition/internal/exec/stages/files.TestMapEntriesToFilesystems(0xc42009af00)
	/home/derek/go/src/github.com/coreos/ignition/gopath/src/github.com/coreos/ignition/internal/exec/stages/files/files_test.go:98 +0xed8
testing.tRunner(0xc42009af00, 0x9358b8)
	/nix/store/1blrc2bxdfjrk3kl46l75dw4i1lpswp5-go-1.7.1/share/go/src/testing/testing.go:610 +0xca
created by testing.(*T).Run
	/nix/store/1blrc2bxdfjrk3kl46l75dw4i1lpswp5-go-1.7.1/share/go/src/testing/testing.go:646 +0x530
```

To reproduce:
- clone https://github.com/dgonyeo/ignition
- `git checkout panic-report`
- `./test`
- `TestMapEntriesToFilesystems` will trigger the panic in this library

**Comment by kylelemons** [comment]
It looks like your stack trace points to a vendored copy of godebug; I believe this issue is fixed in the main repo (by 4da1e5af9387fc77d0e824ea7f04a6a39d6df1d3).  The line numbers in the stack trace also don't reflect the state at head anymore, so if you could reopen if you can reproduce this with the latest version of the library that would be much appreciated, thanks!

**Comment by dgonyeo** [comment]
Ah, you're right. No clue why I wasn't using the latest one. Thanks!
